src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -73,15 +73,18 @@ internal virtual DiagnosticInfo GetConstraintsUseSiteErrorInfo()
         /// Duplicates and cycles are removed, although the collection may include
         /// redundant constraints where one constraint is a base type of another.
         /// </summary>
-        internal ImmutableArray<TypeSymbolWithAnnotations> ConstraintTypesNoUseSiteDiagnostics
+        internal ImmutableArray<TypeSymbolWithAnnotations> GetConstraintTypesNoUseSiteDiagnostics(ConsList<TypeParameterSymbol> inProgress, bool early)
         {
-            get
+            this.EnsureAllConstraintsAreResolved(early: true);
```

Comment:
>EnsureAllConstraintsAreResolved [](start = 17, length = 31)

Is there a difference with just calling `EnsureAllConstraintsAreResolved(early)`? I would have assumed that calling with `early : false` would just do a superset of the work that calling with `early : true` does.

Reply:
The work for `early: false` is a superset of the `early: true`, but the early work should be completed for all type parameters before the late work is started for any type parameter. This split (two separate calls), needs to be done somewhere, and it was simpler to do here, in the caller of `EnsuresAllConstraintsAreResolved` rather than in several implementations of that `abstract` method. (This should be the only caller of that method that support `early: false`.)

---
In reply to: [195540314](https://github.com/dotnet/roslyn/pull/27525#discussion_r195540314) [](ancestors = 195540314)

Reply:
Thanks. Probably worth a comment